slack-plugin/src/main/kotlin/slack/gradle/lint/LintTasks.kt

```diff
@@ -87,139 +255,196 @@ internal object LintTasks {
+      disable.add("SyntheticAccessor")
 
-    slackProperties.ciLintVariants?.let { variants ->
-      ciLintTask.configure {
-        // Even if the task isn't created yet, we can do this by name alone and it will resolve at
-        // task configuration time.
-        variants.splitToSequence(',').forEach { variant ->
-          logger.debug("Using variant $variant for ciLint task")
-          val lintTaskName = "lint${variant.capitalizeUS()}"
-          dependsOn(lintTaskName)
+      // Only check for missing translations in finalized (beta and later) modules.
+      fatal.add("MissingTranslation")
+
+      if (androidExtensionNullable is ApplicationAndroidComponentsExtension) {
+        checkDependencies = true
+      }
+
+      androidSdkVersions?.let { sdkVersions ->
+        if (sdkVersions.minSdk >= 28) {
+          // Lint doesn't understand AppComponentFactory
+          // https://issuetracker.google.com/issues/243267012
+          disable += "Instantiatable"
         }
       }
-      return // nothing else to do!
+
+      lintConfig = project.rootProject.layout.projectDirectory.file("config/lint/lint.xml").asFile
+
+      baseline =
+        project.objects
+          .fileProperty()
+          .fileValue(File(project.projectDir, slackProperties.lintBaselineFileName))
+          .get()
+          .asFile
     }
+  }
 
-    val componentsExtension =
-      when (extension) {
-        is AppExtension -> {
-          project.extensions.getByType<ApplicationAndroidComponentsExtension>()
-        }
-        is LibraryExtension -> {
-          project.extensions.getByType<LibraryAndroidComponentsExtension>()
-        }
-        is TestExtension -> {
-          project.extensions.getByType<TestAndroidComponentsExtension>()
-        }
-        else -> error("No AndroidComponentsExtension found for project ${project.path}")
+  /**
+   * If the project is using multiplatform, adds configurations and source sets expected by the lint
+   * plugin, which allows it to configure itself when running against a non-Android multiplatform
+   * project.
+   *
+   * The version of lint that we're using does not directly support Kotlin multiplatform, but we can
+   * synthesize the necessary configurations and source sets from existing `jvm` configurations and
+   * `kotlinSourceSets`, respectively.
+   *
+   * This method *must* run after evaluation.
+   */
+  private fun Project.addSourceSetsForMultiplatformAfterEvaluate() {
+    val kmpTargets = project.multiplatformExtension?.targets ?: return
+
+    // Synthesize target configurations based on multiplatform configurations.
+    val kmpApiElements = kmpTargets.map { it.apiElementsConfigurationName }
+    val kmpRuntimeElements = kmpTargets.map { it.runtimeElementsConfigurationName }
+    listOf(kmpRuntimeElements to "runtimeElements", kmpApiElements to "apiElements").forEach {
+      (kmpConfigNames, targetConfigName) ->
+      project.configurations.maybeCreate(targetConfigName).apply {
```

Comment:
since we're potentially creating new configurations, should we be setting `isResolvable = false`? (since these are "Elements" configs, they're consumable but not resolvable, and isConsumable defaults to true).

Reply:
I don't know 🙃